rust/otap-dataflow/crates/otlp/src/debug_exporter/exporter.rs

```diff
@@ -0,0 +1,427 @@
+    metric_request: ExportMetricsServiceRequest,
+    marshaler: &dyn PDataMarshaler,
+) {
+    // collect number of resource metrics
+    // collect number of metrics
+    // collect number of datapoints
+    let resouce_metrics = metric_request.resource_metrics.len();
+    let mut data_points = 0;
+    let mut metrics = 0;
+    for resource_metrics in &metric_request.resource_metrics {
+        for scope_metrics in &resource_metrics.scope_metrics {
+            metrics += scope_metrics.metrics.len();
+            for metric in &scope_metrics.metrics {
+                if let Some(data) = &metric.data {
+                    match data {
+                        Data::Gauge(gauge) => {
+                            data_points += gauge.data_points.len();
+                        }
+                        Data::Sum(sum) => {
+                            data_points += sum.data_points.len();
+                        }
+                        Data::Histogram(histogram) => {
+                            data_points += histogram.data_points.len();
+                        }
+                        Data::ExponentialHistogram(exponential_histogram) => {
+                            data_points += exponential_histogram.data_points.len();
+                        }
+                        Data::Summary(summary) => {
+                            data_points += summary.data_points.len();
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    println!("Received {} resource metrics", resouce_metrics);
+    println!("Received {} metrics", metrics);
+    println!("Received {} data points", data_points);
+
+    if *verbosity == Verbosity::Basic {
+        return;
+    }
+
+    let report = marshaler.marshal_metrics(metric_request);
+    println!("{}", report);
+    return;
+}
+
+fn push_trace(
+    verbosity: &Verbosity,
+    trace_request: ExportTraceServiceRequest,
+    marshaler: &dyn PDataMarshaler,
+) {
+    // collect number of resource spans
+    // collect number of spans
+    let resource_spans = trace_request.resource_spans.len();
+    let mut spans = 0;
+    for resource_span in &trace_request.resource_spans {
+        for scope_span in &resource_span.scope_spans {
+            spans += scope_span.spans.len();
+        }
+    }
+    println!("Received {} resource spans", resource_spans);
+    println!("Received {} spans", spans);
+    if *verbosity == Verbosity::Basic {
+        return;
+    }
+
+    let report = marshaler.marshal_traces(trace_request);
+    println!("{}", report);
+
+    return;
+}
+
+fn push_log(
+    verbosity: &Verbosity,
+    log_request: ExportLogsServiceRequest,
+    marshaler: &dyn PDataMarshaler,
+) {
+    let resource_logs = log_request.resource_logs.len();
+    let mut log_records = 0;
+    for resource_log in &log_request.resource_logs {
+        for scope_log in &resource_log.scope_logs {
+            log_records += scope_log.log_records.len();
```

Comment:
You could keep track of 2 counters:
- total number of logs
- total number of logs with a name field defined (events)